src/Umbraco.New.Cms.Core/Services/Installer/InstallService.cs

```diff
@@ -0,0 +1,53 @@
+﻿using Microsoft.Extensions.Logging;
+using Umbraco.Cms.Core;
+using Umbraco.Cms.Core.Services;
+using Umbraco.New.Cms.Core.Installer;
+using Umbraco.New.Cms.Core.Models.Installer;
+
+namespace Umbraco.New.Cms.Core.Services.Installer;
+
+public class InstallService : IInstallService
+{
+    private readonly ILogger<InstallService> _logger;
+    private readonly NewInstallStepCollection _installSteps;
+    private readonly IRuntimeState _runtimeState;
+
+    public InstallService(
+        ILogger<InstallService> logger,
+        NewInstallStepCollection installSteps,
+        IRuntimeState runtimeState)
+    {
+        _logger = logger;
+        _installSteps = installSteps;
+        _runtimeState = runtimeState;
+    }
+
+    /// <inheritdoc/>
+    public async Task Install(InstallData model)
+    {
+        if (_runtimeState.Level != RuntimeLevel.Install)
+        {
+            throw new InvalidOperationException($"Runtime level must be Install to install but was: {_runtimeState.Level}");
+        }
+
+        await RunSteps(_installSteps, model);
+    }
+
+    private async Task RunSteps(IEnumerable<IInstallStep> steps, InstallData model)
```

Comment:
Don't pass the steps as a parameter, `RunSteps` can access the private field `_installSteps`